### PR TITLE
feat(initialisation): Fail gracefully on incorrect targetframework in csproj

### LINKED ISSUE #861: Fail gracefully on incorrect targetframework in csproj
**Describe the bug**
Im running stryker for netcore solution and the project under test target framework is:

<TargetFrameworks>netstandard2</TargetFrameworks>

dotnet stryker --solution-path C:\GIT\DefaultCollection\MyCompany\myService\MySln.sln --project-file Service.csproj --test-project-file UnitTests.csproj --reporters "['html', 'progress']" --excluded-mutations "['string', 'logical']" --coverage-analysis perTest --threshold-high 90 --threshold-low 75 --threshold-break 0 --log-level "debug"

**Logs**
```
[15:52:59 ERR] An error occurred during the mutation test run
System.ArgumentException: Version string portion was too short or too long.
Parameter name: input
   at System.Version.ParseVersion(ReadOnlySpan`1 input, Boolean throwOnFailure)
   at System.Version.Parse(String input)
   at System.Version..ctor(String version)
   at Stryker.Core.Initialisation.ProjectAnalyzerResult.get_TargetFrameworkAndVersion() in d:\a\1\s\src\Stryker.Core\Stryker.Core\Initialisation\ProjectInfo.cs:line 260
   at Stryker.Core.Initialisation.ProjectAnalyzerResult.get_TargetFramework() in d:\a\1\s\src\Stryker.Core\Stryker.Core\Initialisation\ProjectInfo.cs:line 106
   at Stryker.Core.Initialisation.InitialisationProcess.Initialize(StrykerOptions options) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Initialisation\InitialisationProcess.cs:line 43
   at Stryker.Core.StrykerRunner.RunMutationTest(StrykerOptions options, IEnumerable`1 initialLogMessages) in d:\a\1\s\src\Stryker.Core\Stryker.Core\StrykerRunner.cs:line 73
[15:52:59 INF] Time Elapsed 00:00:21.1318803
```

**Expected behavior**
To work?:)

**Desktop (please complete the following information):**
 - OS: Windows
 - Type of project: core
 - Framework Version: 2.2
 - Stryker Version: 0.15.0

**Additional context**
I have debugged your solution and it seems like the issue is here:

image

It takes the "netstandard2" and pulls the "2" from it, then it tries to put it in the "new Version(version)" code.

Please advice.

Thanks,

Greg

### PR DESCRIPTION
**Pull request by dupdob**
Throws an exception with an explicit error message when failing to parse net framework version from the csproj

Fix #861

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs [hunk 1]
```diff
@@ -241,26 +241,33 @@ public IEnumerable<ResourceDescription> Resources
...
                         }
+                        return (label[analysis.Groups["kind"].Value], new Version(version));
                     }
-                    return (label[analysis.Groups["kind"].Value], new Version(version));
-                }
 
-                return (Framework.Unknown, new Version());
+                    return (Framework.Unknown, new Version());
+                }
+                catch (ArgumentException exception)
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs]:
```suggestion
                catch (ArgumentException)
```

File: src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs [hunk 2]
```diff
@@ -241,26 +241,33 @@ public IEnumerable<ResourceDescription> Resources
...
+                {
+                    throw new ApplicationException($"Unable to parse framework version string {TargetFrameworkVersionString}.", exception);
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs]:
```suggestion
                    throw new StrykerInputException($"Unable to parse framework version string {TargetFrameworkVersionString}. Please fix the framework version in the csproj.");
```

**Comment by Mobrockers** [comment]
Oops forgot to add the using. Can't seem to place it as a suggestion. You'll need this using: `using Stryker.Core.Exceptions`

**Comment by dupdob** [comment]
Should have spotted that.

File: src/Stryker.Core/Stryker.Core/TestRunners/VsTest/RunEventHandler.cs [hunk 3]
```diff
@@ -46,6 +46,10 @@ public RunEventHandler(AutoResetEvent waitHandle, ILogger logger, string runnerI
 
             TimeOut = testRunCompleteArgs.IsAborted;
 
+            TimeOut = testRunCompleteArgs.IsAborted;
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/TestRunners/VsTest/RunEventHandler.cs]:
```suggestion
```

File: src/Stryker.Core/Stryker.Core/TestRunners/VsTest/RunEventHandler.cs [hunk 4]
```diff
@@ -46,6 +46,10 @@ public RunEventHandler(AutoResetEvent waitHandle, ILogger logger, string runnerI
...
+
+            TimeOut = testRunCompleteArgs.IsAborted;
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/TestRunners/VsTest/RunEventHandler.cs]:
```suggestion
```

File: src/Stryker.Core/Stryker.Core/TestRunners/VsTest/RunEventHandler.cs (hunk 3 above)
Comment (dupdob) [reply src/Stryker.Core/Stryker.Core/TestRunners/VsTest/RunEventHandler.cs]: Yep
For some reason,  my repo is out of sync and leads to difficult merge
sorry for that,

**Pull request merged**